torch/csrc/autograd/record_function.cpp

```diff
@@ -26,39 +29,85 @@ void popCallback() {
   }
   start_callbacks.pop_back();
   end_callbacks.pop_back();
-  has_callbacks = !start_callbacks.empty();
+  needs_inputs_flags.pop_back();
+  if (needs_inputs_flags.size() > 1) {
+    if (needs_inputs_flags.back()) {
+      record_function_state = RecordFunctionState::NAME_AND_INPUTS;
```

Comment:
FWIW this should never be necessary

Reply:
please refer to https://github.com/pytorch/pytorch/issues/18966 that explain why it is necessary in some use cases